src/safeposix/syscalls/sys_calls.rs

```diff
@@ -128,6 +128,17 @@ impl Cage {
             } else {panic!("We changed from a directory that was not a directory in chdir!");}
         } else {panic!("We changed from a directory that was not a directory in chdir!");}
 
+        // we grab the parent cages main threads sigset and store it at 0
+        // we do this because we haven't established a thread for the cage yet, and dont have a threadid to store it at
+        // this way the child can initialize the sigset properly when it establishes its own mainthreadid
+        let newsigset = interface::RustHashMap::new();
+        if !interface::RUSTPOSIX_TESTSUITE.load(interface::RustAtomicOrdering::Relaxed) { // we don't add these for the test suite
+            let mainsigsetatomic = self.sigset.get(&self.main_threadid.load(interface::RustAtomicOrdering::Relaxed)).unwrap();
+            let mainsigset = interface::RustAtomicU64::new(mainsigsetatomic.load(interface::RustAtomicOrdering::Relaxed));
```

Comment:
I would suggest Error Handling with unwrap: Using unwrap() can cause panics. It's preferable to handle potential errors gracefully.